subgraphs/burrow/src/handlers/actions.ts

```diff
@@ -0,0 +1,589 @@
+      market._totalReserved = BD_ZERO;
+    }
+  }
+
+  // snapshot
+  dailySnapshot.dailyWithdrawUSD = dailySnapshot.dailyWithdrawUSD.plus(
+    withdraw.amountUSD
+  );
+  hourlySnapshot.hourlyWithdrawUSD = hourlySnapshot.hourlyWithdrawUSD.plus(
+    withdraw.amountUSD
+  );
+  financialDailySnapshot.dailyWithdrawUSD =
+    financialDailySnapshot.dailyWithdrawUSD.plus(withdraw.amountUSD);
+
+  const positionSnapshot = getOrCreatePositionSnapshot(position, receipt);
+  positionSnapshot.logIndex = logIndex as i32;
+
+  // update market
+  updateMarket(
+    market,
+    dailySnapshot,
+    hourlySnapshot,
+    financialDailySnapshot,
+    receipt
+  );
+
+  positionSnapshot.save();
+  financialDailySnapshot.save();
+  dailySnapshot.save();
+  hourlySnapshot.save();
+  usageDailySnapshot.save();
+  usageHourlySnapshot.save();
+  market.save();
+  account.save();
+  withdraw.save();
+  position._last_active_timestamp = BigInt.fromU64(
+    receipt.block.header.timestampNanosec / 1000000000
+  );
+  position.save();
+
+  updateProtocol();
+}
+
+/**
+ * Handles borrow events
+ * @param event args {account_id, amount, token_id}
+ * @notice Borrowed tokens are added as supplied tokens. Users need to withdraw tokens to have them in wallet
+ */
+export function handleBorrow(event: EventData): void {
+  const receipt = event.receipt;
+  const logIndex = event.logIndex;
+  const data = event.data;
+
+  const borrow = getOrCreateBorrow(
+    receipt.outcome.id
+      .toBase58()
+      .concat("-")
+      .concat((logIndex as i32).toString()),
+    receipt
+  );
+
+  borrow.logIndex = logIndex as i32;
+  const parsedData = parse0(data);
+  const account_id = parsedData[0];
+  const amount = parsedData[1];
+  const token_id = parsedData[2];
+
+  const market = getOrCreateMarket(token_id.toString());
+  const protocol = getOrCreateProtocol();
+  const dailySnapshot = getOrCreateMarketDailySnapshot(market, receipt);
+  const hourlySnapshot = getOrCreateMarketHourlySnapshot(market, receipt);
+  const usageDailySnapshot = getOrCreateUsageMetricsDailySnapshot(receipt);
+  const usageHourlySnapshot = getOrCreateUsageMetricsHourlySnapshot(receipt);
+  const financialDailySnapshot = getOrCreateFinancialDailySnapshot(receipt);
+  const account = getOrCreateAccount(account_id.toString());
+  const position = getOrCreatePosition(account, market, "BORROWER", receipt);
+
+  const token = getOrCreateToken(token_id.toString());
+  borrow.account = account.id;
+  borrow.amount = BigInt.fromString(amount.toString());
+  borrow.asset = token.id;
+  borrow.market = market.id;
+  borrow.position = position.id;
+
+  borrow.amountUSD = borrow.amount
+    .divDecimal(
+      BigInt.fromI32(10)
+        .pow((token.decimals + token.extraDecimals) as u8)
+        .toBigDecimal()
+    )
+    .times(token.lastPriceUSD!);
+
+  if (position._last_active_timestamp.lt(usageDailySnapshot.timestamp)) {
+    usageDailySnapshot.dailyActiveBorrowers += 1;
```

Comment:
same thing, this actually results in the `dailyActiveBorrowers` == `dailyBorrowCount` which would be pretty unlikely every day

Reply:
will do